src/SuperstateToken.sol

```diff
@@ -112,10 +107,11 @@ abstract contract SuperstateToken is ERC20Upgradeable, IERC7246, PausableUpgrade
     function initialize(string calldata _name, string calldata _symbol) public initializer {
         __ERC20_init(_name, _symbol);
         __Pausable_init();
+        __Ownable2Step_init();
```

Comment:
i think this means deployer has to be the admin? was that intentional? in general imo its better if deployer and admin are not coupled. that way there can be one deployer address that is funded with a lot of testnet tokens. and generally lower friction to deploy, can set tighter policies on the signing keys.

Reply:
its possible to use Ownable to set admin like

`    constructor(address morphoAddress, address owner) Ownable(owner) {`

Reply:
Yeah agree that it's better to have them separate and was trying to do that initially but was having issues trying to actually do that. Will try your suggestion

Reply:
@maxwolff , OTOH, one benefit of this is we must perform a proof of control to accept ownership. That way we can't mistake inputting the admin key

Reply:
you mean deployer deploys, and transfer to signer?

Reply:
Yeah, deployer would deploy w/ itself as the original admin.
Then it broadcasts a `transferOwnership(turnkey_key)`
And then turnkey_key responds wth `acceptOwnership` to get proof of control.